Minesweeper.java

```diff
@@ -118,8 +170,8 @@ private void computeNeighboringBombs(Cell cell) {
                     }
                 }
             }
-
             cell.setNeighboringBombs(neighboringBombs);
+            return neighboringBombs;
```

Comment:
i dont think we should make this function return anything if it is also setting the `neighboringBombs` value for the cell, which means we can just get the value from the cell. 
Then, the code on line 121 would become:
```java
// If this cell has no adjacent mines, recursively reveal its neighbors
computeNeighboringBombs(this.cells[row][col])
if (this.cells[row][col].neighboringBombs == 0) {
  ...
}
```

Alternatively, the `revealEmptyCells` method could be implemented directly inside of `computeNeighboringBombs` since they serve the same function except that `revealEmptyCells` is dedicated to recursion only, while incorporating it into `computeNeighboringBombs` would make the way we reveal cells itself recursive, which in my opinion better aligns with the idea of a recursive method for revealing cells.

This route would also allow us to take advantage of the fact that we have the ability to reveal cells *before* their neighboring bombs are computed, which effectively means we can skip the "this cell is equal to the selected cell" check and only check if the cell is revealed

Reply:
yeah i agree, i was trying a different thing with it but at the end it's really not needed so it could just be a void

Reply:
> yeah i agree, i was trying a different thing with it but at the end it's really not needed so it could just be a void

latest commit addresses this, have a look